OrbitClientModel/include/OrbitClientModel/CaptureSerializer.h

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CLIENT_MODEL_CAPTURE_SERIALIZER_H_
+#define ORBIT_CLIENT_MODEL_CAPTURE_SERIALIZER_H_
+
+#include <iosfwd>
+#include <outcome.hpp>
+#include <string>
+
+#include "CaptureData.h"
+#include "OrbitBase/Result.h"
+#include "capture_data.pb.h"
+#include "google/protobuf/io/coded_stream.h"
+#include "google/protobuf/io/zero_copy_stream_impl.h"
+#include "google/protobuf/message.h"
+
+namespace CaptureSerializer {
+
+template <class TimersIterator>
+ErrorMessageOr<void> Save(const std::string& filename, const CaptureData& capture_data,
+                          const absl::flat_hash_map<uint64_t, std::string>& key_to_string_map,
+                          TimersIterator timers_iterator_begin, TimersIterator timers_iterator_end);
+
+void WriteMessage(const google::protobuf::Message* message,
+                  google::protobuf::io::CodedOutputStream* output);
+
+}  // namespace CaptureSerializer
+
+namespace internal {
```

Comment:
I think you should make this an anonymous namespace (and move the implementation into this header file).
@danielfenner please correct me, if I am wrong.

Reply:
I tried using anonymous and it was not working, did not compile and had linking errors. I made a quick search and apparently anonymous namespaces should not be used in a header. However, I may be wrong

Reply:
I think that's not a good idea. An anonymous namespace only makes sense in a cpp-file. I would stick to the internal-namespace but rather move into the `CaptureSerializer`-namespace because `internal::Save` is too generic. This name could occur somewhere else in the codebase as well but `CaptureSerializer::internal::Save` is fine.

And just like Florian said, I would also move the implementation from below up here. I don't see a need to have a separate declaration in this case.

Reply:
And I'm sorry for chiming in. :see_no_evil:

Reply:
Thanks for clarifying this. So then I would also appreciate a not that generic name. @beckerhe do you have a link or so, why anonymous namespace is not a good idea here?

Reply:
Ok, I will update the code. Thanks a lot for the clarification :)

Reply:
I don't have a link right away. But I think it's pretty clear from what an anonymous namespace does. First of all, it does NOT affect name resolution (that's not a 100% true, but almost). That means for the compiler it's like if there were no namespace. So the `Save` would be in the global namespace. The only thing an anonymous namespace does it changes the linkage of the symbols defined inside to internal. "internal linkage" means the symbols are only visible in the current compilation unit. That means every compilation-unit including this header file will instantiate their own version of this templated function. Instead of having only one copy of that function in the binary we will have as many as we have calls.

Reply:
The best thing I can offer right away is the reference: https://en.cppreference.com/w/cpp/language/namespace#Unnamed_namespaces

Reply:
I have updated the code, thanks!

Reply:
Thanks (both for clarification and for the update)